Automattic/wp-calypso
# Gutenberg: Build a new implementation of Publicize wp.data store

#### Changes proposed in this Pull Request

* Introduce a new Publicize store implementation with selectors and resolvers
* Update extension to use the new store
* Remove the old store and library for requests

#### Testing instructions

* Spin up a site with this branch - https://jurassic.ninja/create?shortlived&gutenpack&gutenberg&calypsobranch=update/gutenberg-publicize-update-wp-data&branch=try/publicize-gutenberg-block-externally-built-rebased
* Uncomment this line: https://github.com/Automattic/wp-calypso/blob/f68ef27219b5818284f08d8f41f3865bd3222fc7/client/gutenberg/extensions/presets/jetpack/editor.js#L9
* Write a new post.
* Attempt to publish the post.
* Verify the Publicize block works like it did before in the pre-publish panel:
  * Try with one or more connections, verify they load properly.
  * Try with no connections, verify you can properly see the list of services available for connecting.
    * Try connecting a service from the list, verify it leads you to the settings page for connecting a service.
  * Try refreshing connections when you have and when you don't have connections.

Fixes #28100
